File: backend/Admin/src/main/java/com/example/entities/Subscription.java
package com.example.entities;

import java.time.LocalDate;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "subscription")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Subscription {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "subscription_id")
    private int subscriptionId;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "plan_id")
    private MonthlyPlan plan;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    private String status;
}

File: backend/Customer/Customer/Models/Subscription.cs
using System;
using System.Collections.Generic;

namespace Customer.Models;

public partial class Subscription
{
    public int SubscriptionId { get; set; }

    public int UserId { get; set; }

    public int PlanId { get; set; }

    public DateOnly StartDate { get; set; }

    public DateOnly EndDate { get; set; }

    public string Status { get; set; } = null!;

    public virtual ICollection<CustomerVisitLog> CustomerVisitLogs { get; set; } = new List<CustomerVisitLog>();

    public virtual MonthlyPlan Plan { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}
